src/profile/useProfile.ts

```diff
@@ -38,41 +38,40 @@ type ProfileSaveCallback = ({
   removeAvatar: boolean;
 }) => Promise<void>;
 
-export function useProfile(client: MatrixClient) {
-  const [{ loading, displayName, avatarUrl, error, success }, setState] =
+export function useProfile(client?: MatrixClient) {
```

Comment:
```suggestion
export function useProfile(client: MatrixClient | undefined) {
```

Let's do this so the caller can't completely forget to pass something for the `client` argument

Reply:
That makes sense, agree! Fixed.